src/cunumeric/set/unique_reduce_template.inl

```diff
@@ -41,33 +54,65 @@ struct UniqueReduceImpl {
       auto shape = input_arr.shape<1>();
       res_size += shape.hi[0] - shape.lo[0] + 1;
     }
-    auto result  = output.create_output_buffer<VAL, 1>(Point<1>(res_size));
-    VAL* res_ptr = result.ptr(0);
 
-    size_t offset = 0;
-    for (auto& input_arr : input_arrs) {
-      size_t strides[1];
-      Rect<1> shape     = input_arr.shape<1>();
-      size_t volume     = shape.volume();
-      const VAL* in_ptr = input_arr.read_accessor<VAL, 1>(shape).ptr(shape, strides);
-      assert(shape.volume() <= 1 || strides[0] == 1);
-      thrust::copy(exe_pol, in_ptr, in_ptr + volume, res_ptr + offset);
-      offset += volume;
-    }
-    assert(offset == res_size);
+    // Is splitting into two completely distinct cases the most concise way to do this?
+    if (return_index) {
+      auto result = output.create_output_buffer<ZippedIndex<VAL>, 1>(Point<1>(res_size));
+      ZippedIndex<VAL>* res_ptr = result.ptr(0);
 
-    thrust::sort(exe_pol, res_ptr, res_ptr + res_size);
-    VAL* actual_end = thrust::unique(exe_pol, res_ptr, res_ptr + res_size);
-    output.bind_data(result, Point<1>(actual_end - res_ptr));
+      size_t offset = 0;
+      for (auto& input_arr : input_arrs) {
+        size_t strides[1];
+        Rect<1> shape = input_arr.shape<1>();
+        size_t volume = shape.volume();
+        const ZippedIndex<VAL>* in_ptr =
+          input_arr.read_accessor<ZippedIndex<VAL>, 1>(shape).ptr(shape, strides);
+        assert(shape.volume() <= 1 || strides[0] == 1);
+        thrust::copy(exe_pol, in_ptr, in_ptr + volume, res_ptr + offset);
+        offset += volume;
+      }
+      assert(offset == res_size);
+
+      thrust::sort(exe_pol, res_ptr, res_ptr + res_size, ZippedComparator<VAL>());
+      ZippedIndex<VAL>* actual_end =
+        thrust::unique(exe_pol, res_ptr, res_ptr + res_size, IndexFreeEqual<VAL>());
```

Comment:
does thrust unique use the ordering policy between indices that we want? if so, would be good to note that here.

Reply:
I had originally not commented that because it was uncommented in the original GPU variant. Would it be best to comment in both places?

Reply:
I don't think that it matters for the original GPU variant that was just doing unique without indices because it didn't matter which of the repeats was chosen. It matters now given that the unique spec says we should take the first one.